sdk/core/azure-core/test/ut/CMakeLists.txt

```diff
@@ -34,6 +35,9 @@ add_executable (
      uuid.cpp
      context.cpp)
 
-target_link_libraries(${TARGET_NAME} PRIVATE azure-core)
-add_gtest(${TARGET_NAME})
+target_link_libraries(${TARGET_NAME} PRIVATE azure-core gtest)
+
+# gtest_add_tests will scan the test from azure-core-test and call add_test
+# for each test to ctest. This enables `ctest -r` to run specific tests directly.
+gtest_add_tests(TARGET ${TARGET_NAME})
```

Comment:
Make sure to update contributor docs to mention this there too (in a separate PR) so folks know how to run single tests.

In general, avoid unrelated changes to keep the PR focused on the issue at hand. Makes reverting and following code change history easier, especially months in the future.